velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -53,15 +60,89 @@ void castFromTimestamp(
   });
 }
 
+void castFromString(
+    const SimpleVector<StringView>& inputVector,
+    exec::EvalCtx& context,
+    const SelectivityVector& rows,
+    VectorPtr& result) {
+  auto timestampWithTzVector = result->asFlatVector<int64_t>();
+  auto rawTimestampWithTzValues =
+      timestampWithTzVector->values()->asMutable<int64_t>();
+  timestampWithTzVector->clearNulls(rows);
+
+  context.applyToSelectedNoThrow(rows, [&](auto row) {
+    if (inputVector.isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      auto tsTzStr = std::string_view(
+          inputVector.valueAt(row).data(), inputVector.valueAt(row).size());
+
+      // Find the second space, as that marks the start
+      // of the timezone component.
+      auto pos = tsTzStr.find(" ");
```

Comment:
Nit : Move this code to TimestampConversion.cpp for more reusable function.

Reply:
You made `splitTimestampWithTimezoneString` so should be used here.